warehouse/router/upload.go

```diff
@@ -466,12 +513,37 @@ func (job *UploadJob) cleanupObjectStorageFiles() error {
 	loadingKeysToDel := lo.Map(loadingFiles, func(file model.LoadFile, _ int) string {
 		return fm.GetDownloadKeyFromFileLocation(file.Location)
 	})
-	if err = fm.Delete(job.ctx, append(stagingKeysToDel, loadingKeysToDel...)); err != nil {
+
+	// GCS doesn't support batch delete, so we need to delete files in chunks to speed up the deletion
+	if storageProvider == whutils.GCS {
+		err = job.deleteFilesInChunks(
+			append(stagingKeysToDel, loadingKeysToDel...),
+			fm,
+			job.config.maxParallelObjDeleteRequests(job.upload.WorkspaceID, storageProvider),
+			job.config.objDeleteBatchSize(job.upload.WorkspaceID, storageProvider),
+		)
+	} else {
+		err = fm.Delete(job.ctx, append(stagingKeysToDel, loadingKeysToDel...))
+	}
```

Comment:
Can we also add some stats (with object storage and workspace information) around this part? 
1. How many files?
2. How much time is it taking?

Reply:
1. Added gauge
2. Added time taken for overall deletion